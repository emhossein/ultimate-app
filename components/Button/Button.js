import React from 'react';
import Link from 'next/link';
import styled from '@emotion/styled';
import { useTheme } from '@emotion/react';

const Button = ({ children, link, onClick }) => {
  const theme = useTheme();

  const Btn = styled.button`
    margin-top: 2rem;
    width: 200px;
    height: 45px;
    border: 1px solid ${theme.colors.primary};
    background: ${link ? theme.colors.primary : theme.colors.secondary};
    color: ${link ? theme.colors.secondary : theme.colors.primary};
    border-radius: 8px;
    transition: all 0.1s;
    cursor: pointer;

    button {
      background: transparent;
      border: 0;
    }

    a {
      text-decoration: none;
      color: ${theme.colors.secondary};
      &:hover {
        color: ${theme.colors.primary};
      }
    }

    &:hover {
      background: ${link ? theme.colors.secondary : theme.colors.primary};
      color: ${link ? theme.colors.primary : theme.colors.secondary};
    }
  `;

  if (!link) {
    return <Btn onClick={onClick}>{children}</Btn>;
  }

  return (
    <Btn type={link ? null : 'submit'}>
      <Link href={`/${link ? link : null}`}>{children}</Link>
    </Btn>
  );
};

export default Button;
