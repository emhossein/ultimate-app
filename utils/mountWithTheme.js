import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@emotion/react';
import theme from '../config/theme';

const mountWithTheme = ({ children }) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  );
};

export default mountWithTheme;
