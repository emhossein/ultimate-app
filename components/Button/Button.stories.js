import React from 'react';

import Button from './Button';

export default {
  title: 'Button',
  component: Button,
};

export const primary = () => <Button link="/">Primary</Button>;
export const secondary = () => <Button>Secondary</Button>;
