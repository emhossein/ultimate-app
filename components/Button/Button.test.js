import React from 'react';
import Button from './Button';
import mountWithTheme from '../../utils/mountWithTheme';

describe('Button', () => {
  test('Render correctly', () => {
    const component = mountWithTheme(<Button>sample button</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  console.log('describe outer-c');
});
