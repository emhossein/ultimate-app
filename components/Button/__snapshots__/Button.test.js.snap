// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-17lxpr6{margin-top:2rem;width:200px;height:45px;border:1px solid #ff846e;background:#fff;color:#ff846e;border-radius:8px;-webkit-transition:all 0.1s;transition:all 0.1s;cursor:pointer;}.css-17lxpr6 button{background:transparent;border:0;}.css-17lxpr6 a{-webkit-text-decoration:none;text-decoration:none;color:#fff;}.css-17lxpr6 a:hover{color:#ff846e;}.css-17lxpr6:hover{background:#ff846e;color:#fff;}",
      }
    }
    data-emotion-css="17lxpr6"
  />,
  <button
    className="css-17lxpr6"
  >
    sample button
  </button>,
]
`;
